### PR TITLE
Implement 'hide zero balance token' setting for token balances on home screen

### LINKED ISSUE #2415: Option: Hide zero balance tokens
This feature was recently added to the extension. Can we get it added to the app?
General > Hide tokens with zero balance

(actual work completed by @darkwing!)

### PR DESCRIPTION
**Pull request by darkwing**
**Description**

This new setting a feature we've added to the MetaMask extension -- hiding tokens whose balances are 0.  This is a significant usability improvement as it reduces clutter.

**Issue**

Resolves #2415

### CODE REVIEW

**Comment by rickycodes** [comment]
looks like snapshots need to be updated:
```
yarn run test:unit -u
```

**Comment by rickycodes** [comment]
also looks like there's a vulnerability that could be patched in another PR :thinking:

I'll take a look at doing that now.

**Review by sethkfman — COMMENTED** [review]
I think this looks good. I would wait on one other mobile team member since I am new.

File: app/components/Views/Settings/GeneralSettings/index.js [hunk 1]
```diff
@@ -249,6 +274,20 @@ class Settings extends PureComponent {
 							)}
 						</View>
 					</View>
+					<View style={styles.setting}>
+						<Text style={styles.title}>Hide Zero-Balance Tokens</Text>
```
Comment (darkwing) [inline app/components/Views/Settings/GeneralSettings/index.js:278]: Needs localization.

File: app/components/Views/Settings/GeneralSettings/index.js [hunk 2]
```diff
@@ -249,6 +274,20 @@ class Settings extends PureComponent {
...
+						<Text style={styles.desc}>
+							Prevents tokens with no balance from displaying in your token listing.
```
Comment (darkwing) [inline app/components/Views/Settings/GeneralSettings/index.js:280]: Needs localization.

**Comment by wachunei** [comment]
Some spacing is missing here, use other setting like hex data as reference.
Screen Shot 2021-03-29 at 16 04 17

Screen Shot 2021-03-29 at 16 04 23

**Review by rickycodes — APPROVED** [review]
LGTM

**Review by ibrahimtaveras00 — APPROVED** [review]
Works as expected, QA Passed 👍🏽

**Pull request merged**